docs/gradle-development.rst

```diff
@@ -0,0 +1,96 @@
+Java components (Gradle)
+========================
+
+.. _Semantic Versioning: https://semver.org
+.. _Git: https://git-scm.com/
+.. _Gradle: https://gradle.org/
+
+This document describes the conventions and process used by the OME team for developing,
+maintaining and releasing its Java components using Gradle_ as their build system. The
+set of rules and procedures described below applies to all the submodules of
+https://github.com/ome/omero-build as well as https://github.com/ome/omero-gradle-plugins/.
+
+.. seealso::
+
+    :doc:`java-development`
+       Conventions and process for maintaining the OME Java Components using Maven
+
+Conventions
+-----------
+
+Source code and build system
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The source code of the components is maintained under version control
+using Git_ and hosted on GitHub_.
+
+Gradle_ is the primary build system. The directory layout should
+follow the standard Maven layout i.e. in the case of a single-module project::
+
+   .github/
+     workflows/     # GitHub actions workflow
+   src/
+     main/          # Component source
+   test/
+     main/
+  CHANGELOG.md
+  LICENSE.txt
+  README.md
+  build.gradle
+  settings.gradle
+
+Additionally, a :file:`publish.gradle` might exist allowing to declare some publishing tasks.
+
+.. seealso::
+   :doc:`using-git`
+
+Development
+^^^^^^^^^^^
+
+Server components follow `Semantic Versioning`_ i.e. given a version number
+MAJOR.MINOR.PATCH, increment the:
+
+- MAJOR version when you make incompatible API changes,
+- MINOR version when you add functionality in a backwards-compatible manner,
+  and
+- PATCH version when you make backwards-compatible bug fixes.
+
+In between releases the version is bumped to the next SNAPSHOT version.
+
+Code contributions should follow the guidelines highlighted in :doc:`code-contributions`.
+
+Distribution
+^^^^^^^^^^^^
+
+All the release artifacts for the repositories listed above should be deployed
+to the OME Artifactory according to the process described in the next
+section.
+
+Release process
+---------------
+
+To  make a new component release, after merging all contributions, on the `master`
+branch, the version defined in :file:`build.gradle` must be moved out of SNAPSHOT
+and set to the target version. Release notes described the major changes should also
+be added to :file:`CHANGELOG.md`.
+
+After committing the changes, a PGP-signed tag must be created for the released version
```

Comment:
This needs to be unified with https://github.com/ome/ome-contributing/pull/8/. One indicates ``GPG`` one ``PGP`` cc @will-moore

Reply:
The usage of `PGP` is historical

```
(base) sbesson@Sebastiens-MacBook-Pro ome-contributing % git grep PGP
docs/ansible-development.rst:A PGP-signed tag of form `x.y.z` should be created for the released version
docs/cpp-development.rst:- have a valid PGP key for signing the tags
docs/cpp-development.rst:A PGP-signed tag should be created for the released version e.g.
docs/gradle-development.rst:After committing the changes, a PGP-signed tag must be created for the released version
docs/java-development.rst:- have a valid PGP key for signing the tags and the JARs
docs/java-development.rst:Additionally, a PGP-signed tag should be created for the released version e.g.
docs/tools-maintenance.rst:- have a valid PGP key for signing the tags
docs/using-git.rst:If you have a PGP key for signing commits and tags, you may want to add it as
docs/using-git.rst:    git config --global user.signingkey YOUR_PGP_KEY_ID
```

and effectively inherited from as far as the OME Trac wiki days. 

No objection to switching to GPG everywhere as this is effectively the technology we are using these days for commit/artifact signatures.